data/src/main/java/com/linkedin/data/schema/IndentedPdlBuilder.java

```diff
@@ -124,6 +120,15 @@ boolean writeDoc(String doc) throws IOException
     return false;
   }
 
+  @Override
+  PdlBuilder writeJson(Object value) throws IOException
+  {
+    DefaultPrettyPrinter prettyPrinter = new DefaultPrettyPrinter();
+    prettyPrinter.indentObjectsWith(new DefaultIndenter(" ", DefaultIndenter.SYS_LF + getIndentSpaces()));
+    _jsonCodec.setPrettyPrinter(prettyPrinter);
```

Comment:
Suggest creating a local codec for this. 
This way of changing the state of a member variable for each function call is not clean.

Reply:
that is why I changed static to property, this is need to add pdl stateful indentation at the beginning of each line in json

Reply:
The state is not for the entire PdlBuilder, but only when writeJson is called (ie at a function scope). After this function is completed, the state of _jsonBuilder is invalid. So using a member variable to set the state for a particular function is odd. 

What if someone uses the _jsonCodec on some other code path without updating the prettyPrinter? 
   It is not explicit that pretty printer should be updated before the code should be used.
What happens if PdlBuilder stops using _jsonCodec in toJson? 
  It is not explict that PdlBuilder is expecting json to be formatted by custom builders.